yt_dlp/downloader/fragment.py

```diff
@@ -21,6 +23,77 @@
 )
 
 
+class Timer:
+    TIMER_FUNC = time.monotonic
+
+    def __init__(self):
+        self._last_ts = self.TIMER_FUNC()
+
+    def elapsed_seconds(self) -> float:
+        return self.TIMER_FUNC() - self._last_ts
+
+    def has_elapsed(self, seconds: float) -> bool:
+        assert seconds > 0.0
+        elapsed_seconds = self.elapsed_seconds()
+        if elapsed_seconds < seconds:
+            return False
+
+        self._last_ts += elapsed_seconds - elapsed_seconds % seconds
```

Comment:
Even in the case of a late call, we would want have the next call be the set time apart and not moved closer
```
Norm    |    |    |    |    |
Current |    |    |      |  |
Desired |    |    |      |    |
```
This ensures we have at minimum x seconds inbetween. But I guess it could be argued for either side. Why did you choose to do it this way?